### ISSUE TITLE
[Snackbar] Can't change top and bottom padding of snackbar's text view

### ISSUE THREAD

**Issue by MarkNjunge** [material-components/material-components-android#1179]
**Description:**
Setting `<item name="android:padding">0dp</item>` changes the start and end padding but the top and bottom remain unchanged.
Setting `android:paddingTop` and `android:paddingBottom` has the same effect.  

![qemu-system-x86_64_QcZZwlDmeT](https://user-images.githubusercontent.com/19492506/78373250-06bfbe00-75d3-11ea-8eda-9566bdbb370a.png)  
From the layout inspector
![studio64_w5ajHNfAxV](https://user-images.githubusercontent.com/19492506/78373325-235bf600-75d3-11ea-8288-d82e9a124cad.png)

**Expected behavior:**
The top and bottom padding change, similar to start and end.

**Source code:**
```xml
<style name="AppTheme" parent="Theme.MaterialComponents.Light.NoActionBar">
    <item name="snackbarTextViewStyle">@style/AppTheme.Snackbar.TextView</item>
</style>

<style name="AppTheme.Snackbar.TextView" parent="@style/Widget.MaterialComponents.Snackbar.TextView">
    <item name="android:padding">0dp</item>
</style>
```

**Android API version:** 29

**Material Library version:** 1.2.0-alpha05

**Device:** Android Emulator - Pixel 2

**Comment by hunterstich** [comment]
You can set your theme's `snackbarStyle` with `android:paddingTop` and `android:paddingBottom` to achieve increase the padding.

```
<style name="AppTheme" parent="Theme.MaterialComponents.DayNight">
    <item name="snackbarStyle">@style/Widget.MyApp.Snackbar</item>
</style>

<style name="Widget.MyApp.Snackbar" parent="@style/Widget.MaterialComponents.Snackbar">
    <item name="android:layout_margin">45dp</item>
    <item name="android:padding">32dp</item>
</style>
```

![image](https://user-images.githubusercontent.com/7459254/78574459-06543c80-77f8-11ea-984e-e7c7015bc11c.png)


Are you trying to remove all top and bottom padding?

**Comment by MarkNjunge** [comment]
Yes, I'm trying to remove it.

**Comment by DKKovalev** [comment]
Hi, i have the same issue, i want to remove extra top and bottom padding from text when it has more than one line:
<img width="410" alt="изображение" src="https://user-images.githubusercontent.com/5726643/78981382-7cbeaa80-7b28-11ea-8ed4-1a34e198da6f.png">

**Comment by DanielMartinus** [comment]
Same here indeed. Can't remove the padding

<img src="https://user-images.githubusercontent.com/1636897/83031379-c6aa1400-a034-11ea-9189-f677c6ec5004.png" width=300/>

There are huge paddings in the TextView in the screenshot above, I tried removing them with `Widget.MaterialComponents.Snackbar.TextView` but no luck. 

```xml
<style name="Components.Snackbar.TextView" parent="Widget.MaterialComponents.Snackbar.TextView">
    <item name="android:textAppearance">@style/TextAppearance.Body2.Regular</item>
    <item name="android:textColor">@color/colorWhite</item>
    <item name="android:layout_marginEnd">8dp</item>
    <item name="lineHeight">20dp</item>
    <item name="android:padding">0dp</item>
    <item name="android:paddingStart">0dp</item>
    <item name="android:paddingEnd">0dp</item>
</style>
```

**Comment by DanielMartinus** [comment]
I just went through the source and found a way of updating the padding of the Snackbar. I'll explain the steps that I went through:

## Snackbar.java

In **Snackbar.java** on line167 you can see it inflating one of two layouts:
`R.layout.mtrl_layout_snackbar_include : R.layout.design_layout_snackbar_include`

In those layouts you see almost two similar layouts, just different stylings and references. What they have in common though is the TextView and Button. The TextView is named:       **android:id="@+id/snackbar_text"**

However this snackbar_text isn't used anywhere in snackbar.java but after some more investigation I found it's used in `SnackbarContentLayout` references all over the place, but most importantly on line 163 in snackbar.java.

```kotlin
final SnackbarContentLayout content =
        (SnackbarContentLayout)
            inflater.inflate(
                hasSnackbarContentStyleAttrs(parent.getContext())
                    ? R.layout.mtrl_layout_snackbar_include
                    : R.layout.design_layout_snackbar_include,
                parent,
                false);
```

## SnackbarContentLayout

Going into this SnackbarContentLayout I looked for `snackbar_text` which is assigned to `messageView`. Following this message view you quickly find line124 where it's doing a condition if the textview contains a single or multilines updating the bottom and top padding.

Following back where `updateViewsWithinLayout` is called from (line101) you see it's choosing between a multiLineVPadding or singleLineVPadding. 

Following `multiLineVPadding` you'll see it assigned on line91 

```Kotlin
final int multiLineVPadding =
        getResources().getDimensionPixelSize(R.dimen.design_snackbar_padding_vertical_2lines);
```

Finally looking up `design_snackbar_padding_vertical_2lines` you'll see it has 24dp assigned.

## Tl;dr

The padding of the Snackbar varies between singleline (14dp) and multiline (24dp). You need to somehow override this. This can be done by overriding this value in your `dimens.xml` for example.

```xml
<dimen name="design_snackbar_padding_vertical_2lines">14dp</dimen>
```

**Comment by bhavindhodia** [comment]
This Works for me to add Margin as well as Padding to Snackbar
**container_snackbar.xml**
```xml
<?xml version="1.0" encoding="utf-8"?>
<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
    <!--    Set Margin Here -->
    <item
        android:left="20dp"
        android:right="20dp"
        android:bottom="10dp"
        android:top="10dp">

        <!-- Insert your shape here: -->
        <shape android:shape="rectangle"  >
            <solid android:color="#FE4C4C" />

            <!-- Padding for inner text-->
            <padding android:left="25dp" android:right="10dp" android:bottom="10dp" android:top="10dp" />
            <corners android:radius="5dp" />

        </shape>
    </item>
</layer-list>
```

And then from Activity  set that Drawable

**MainActivity.java**
```java
Snackbar snack = Snackbar
                 .make(activity,"Hello World 🚀",Snackbar.LENGTH_INDEFINITE);
        snack.getView()
        .setBackground(ContextCompat.getDrawable(getApplicationContext(),R.drawable.contianer_snackbar));
        snack.show();
```


[Result][1]
 
  [1]: https://i.stack.imgur.com/uTh7k.png

**Comment by gabrielemariotti** [comment]
With `1.2.0`-`1.2.1` and `1.3.0-alpha02`:

    <style name="AppTheme" parent="Theme.MaterialComponents.DayNight">
        <item name="snackbarStyle">@style/Widget.MyApp.Snackbar</item>
    </style>

    <style name="Widget.MyApp.Snackbar" parent="@style/Widget.MaterialComponents.Snackbar">
        
        <item name="android:layout_margin">45dp</item>
        
        <item name="android:padding">32dp</item>
    </style>

<img width="313" alt="Schermata 2020-09-07 alle 09 05 23" src="https://user-images.githubusercontent.com/2583078/92358220-7b020e80-f0e9-11ea-97c0-002ab21f8617.png">
<img width="323" alt="Schermata 2020-09-07 alle 09 08 12" src="https://user-images.githubusercontent.com/2583078/92358371-ae449d80-f0e9-11ea-99d9-1ffc150f1516.png">

**Comment by tfcporciuncula** [comment]
There's a separate issue now specifically about the spacing difference depending on the number of lines of text: https://github.com/material-components/material-components-android/issues/1721

**Comment by fr-jadengu** [comment]
> With `1.2.0`-`1.2.1` and `1.3.0-alpha02`:
> 
> ```
> <style name="AppTheme" parent="Theme.MaterialComponents.DayNight">
>     <item name="snackbarStyle">@style/Widget.MyApp.Snackbar</item>
> </style>
> 
> <style name="Widget.MyApp.Snackbar" parent="@style/Widget.MaterialComponents.Snackbar">
>     
>     <item name="android:layout_margin">45dp</item>
>     
>     <item name="android:padding">32dp</item>
> </style>
> ```
> 
> <img alt="Schermata 2020-09-07 alle 09 05 23" width="313" src="https://user-images.githubusercontent.com/2583078/92358220-7b020e80-f0e9-11ea-97c0-002ab21f8617.png">
> 
> <img alt="Schermata 2020-09-07 alle 09 08 12" width="323" src="https://user-images.githubusercontent.com/2583078/92358371-ae449d80-f0e9-11ea-99d9-1ffc150f1516.png">

Looks it only works for that horizontally, no effect for vertical. I'm using `1.2.1`.

**Comment by vineelk8** [comment]
> ```
> android:layout_marginEnd
> ```

Hi,

But the text inside the snack bar is not aligned with the snack bar. How can I achieve it

**Comment by vineelk8** [comment]
> This Works for me to add Margin as well as Padding to Snackbar **container_snackbar.xml**
> 
> ```
> <?xml version="1.0" encoding="utf-8"?>
> <layer-list xmlns:android="http://schemas.android.com/apk/res/android">
>     
>     <item
>         android:left="20dp"
>         android:right="20dp"
>         android:bottom="10dp"
>         android:top="10dp">
> 
>         
>         <shape android:shape="rectangle"  >
>             <solid android:color="#FE4C4C" />
> 
>             
>             <padding android:left="25dp" android:right="10dp" android:bottom="10dp" android:top="10dp" />
>             <corners android:radius="5dp" />
> 
>         </shape>
>     </item>
> </layer-list>
> ```
> 
> And then from Activity set that Drawable
> 
> **MainActivity.java**
> 
> ```java
> Snackbar snack = Snackbar
>                  .make(activity,"Hello World 🚀",Snackbar.LENGTH_INDEFINITE);
>         snack.getView()
>         .setBackground(ContextCompat.getDrawable(getApplicationContext(),R.drawable.contianer_snackbar));
>         snack.show();
> ```
> 
> [Result](https://i.stack.imgur.com/uTh7k.png)

Hi,

But the text inside the snack bar is not aligned with the snack bar. How can I achieve it
